### DISCUSSION TITLE
postCreateCommand doesn't work.

### DISCUSSION THREAD

**Discussion by deepansh96** [community/community#10480] [Codespaces]
We are a team of 5 people trying out codespaces for our repos. We have three repos in our organization that we need to work across.

Assuming the names of the repos is `frontend`, `backend` and `analytics`, 
We all started a codespace from the `frontend` repo and cloned the other two repos and added them to the vscode workspace.

The issue:
Because we started the codespaces from the `frontend` repo, the github token generated is scoped to the frontend repo. In the codespaces workspace, we cannot push changes to other repos `backend` and `analytics`.

Tried this solution:
We created a Personal Access Token for each of us individually, added them as secrets from the menu "Codepaces secrets". 
We are able to access the PAT when logging into our workspace and can do `export GITHUB_TOKEN = $PAT`. This lets us push changes to the other repos.

But we want to automate it.
We tried
```json
{
  "postCreateCommand": "export GITHUB_TOKEN = $PAT"
}
```
but this isn't working. Also, I have no idea where to see the logs for this. I know the command doesn't run because the github token is still the old, scoped one and not the Personal Access Token that we created.

Any help is appreciated. Thank you!

**Comment by joshspicer** [comment]
I'd suggest not trying to overwrite the `GITHUB_TOKEN` environment variable we provide, and instead create a shell script that executes on `postCreate`.

We have a template that does a similar thing (clones an Azure DevOps git repo in the postCreate).  Here we add a new remote source with `git remote` and the PAT, and then do the `git clone`.  I think something similar may work well for your use-case!

https://github.com/vsls-contrib/ado-in-codespaces/blob/main/.codespaces/init-ado-repo.sh#L199-L211

For GitHub I expect the clone statement would look something like this: 

```
git clone https://username:$PAT@github.com/<owner>/<repo>.git
```

**Comment by bartr** [comment]
The reason it's not doing what you want is post-create runs in a different shell.  Here's how we do it:

```bash

{
    echo "if [ \"\$PAT\" != \"\" ]"
    echo "then"
    echo "    export GITHUB_TOKEN=\$PAT"
    echo "fi"

} >> "$HOME/.zshrc"

```

We use zsh, so if you use bash, change to .bashrc

If you have a Codespaces secret of PAT, it gets set. If not, it doesn't. Works really well for multi-repo and for ghcr.io images.

Make sure you escape the $ or your PAT will get checked in and instantly cancelled (yes, I have - more than once :)

Note that depending on when this runs, you may have to create a new shell to get the PAT. We put it in on-create and use pre-builds to solve that. I've also gotten in the habit of exiting that first shell "just in case".
